sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/LogsHelper.cs

```diff
@@ -22,8 +20,22 @@ namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
     internal static class LogsHelper
     {
         private const int Version = 2;
-        private static readonly ConcurrentDictionary<int, string> s_depthCache = new ConcurrentDictionary<int, string>();
-        private static readonly Func<int, string> s_convertDepthToStringRef = ConvertDepthToString;
+        private static readonly Action<LogRecordScope, IDictionary<string, string>> s_processScope = (scope, properties) =>
+        {
+            foreach (KeyValuePair<string, object?> scopeItem in scope)
+            {
+                if (string.IsNullOrEmpty(scopeItem.Key) || scopeItem.Key == "{OriginalFormat}")
+                {
+                    continue;
+                }
+
+                if (scopeItem.Value != null)
+                {
+                    properties.Add(scopeItem.Key,
+                                    Convert.ToString(scopeItem.Value, CultureInfo.InvariantCulture) ?? "null");
```

Comment:
You're already checking for `scopeItem.Value` to not be `null`.

Is this `?? "null"` still needed?

Reply:
This could actually end up calling the user's custom implementation of the `ToString` method which could not only return a `null` but also throw an exception.

Do we have a mechanism to guard against this scenario?

Reply:
It would be good to add a test case to cover when a user's implementation can throw.

Reply:
Fixed dc9f3376dfb569d744c342e19aa06a874424f2e0